### DISCUSSION TITLE
how to use watermelon db with zustand?

### DISCUSSION THREAD

**Discussion by leamandeep** [pmndrs/zustand#2209] [Q&A] [answered]
I'm trying to use Zustand with watermelon db. I have implemented it. But the problem is whenever I load the app, it is freshly loaded. Zustand starts with initialvalue. I don't know what is the right way to do it?

Here's the watermelon docs
https://watermelondb.dev/docs

Here's my code
```ts
export const useAppStore = create<AppStoreType>((set) => ({
    accessToken: '',
    userId: '',
    setToken: async (token) => {
        await database.localStorage.set("access_token", token)
        set({ accessToken: await database.localStorage.get("access_token") })
    },
    removeToken: async () => {
        await database.localStorage.remove("access_token")
        set({ accessToken: '' })
    },
    setId: async (id) => {
        await database.localStorage.set("user_id", id)
        set({ userId: await database.localStorage.get("user_id") })
    },
    removeId: async () => {
        await database.localStorage.remove("user_id")
        set({ userId: '' })
    }

}))
```

**Comment by dbritto-dev** [comment, marked as answer]
@elamandeep what about using custom storage -> https://docs.pmnd.rs/zustand/integrations/persisting-store-data#how-can-i-use-a-custom-storage-engine

**Reply by leamandeep** [reply]
@dbritto-dev Thanks. Pardon me for asking so many question. Can u guide me on this question. How I can synchronize my database with Zustand? https://watermelondb.dev/docs/Sync/Frontend

**Reply by dbritto-dev** [reply]
@elamandeep do you still need an example?

**Reply by leamandeep** [reply]
No, i don't need example😄. I'm just asking about this https://watermelondb.dev/docs/Sync/Frontend. My question can it be possible to implement watermelon sync feature in zustand?

**Reply by dbritto-dev** [reply]
@elamandeep you can use subscribe to do that
